Pod/Classes/PINAnimatedImageManager.m

```diff
@@ -471,10 +552,16 @@ + (void)writeFileHeader:(NSFileHandle *)fileHandle width:(UInt32)width height:(U
  [frame data] width * height * 4 bytes
  */
 
-+ (void)writeFrameToFile:(NSFileHandle *)fileHandle duration:(Float32)duration frameData:(NSData *)frameData
++ (NSError *)writeFrameToFile:(NSFileHandle *)fileHandle duration:(Float32)duration frameData:(NSData *)frameData
 {
-  [fileHandle writeData:[NSData dataWithBytes:&duration length:sizeof(duration)]];
-  [fileHandle writeData:frameData];
+  NSError *error = nil;
+  @try {
+    [fileHandle writeData:[NSData dataWithBytes:&duration length:sizeof(duration)]];
+    [fileHandle writeData:frameData];
```

Comment:
No, we leave it open as this is only writing one frame. It will be closed once the other frames are written.